PyInstaller/building/build_main.py

```diff
@@ -374,10 +374,28 @@ def assemble(self):
         if is_win:
             depmanifest.writeprettyxml()
 
-        # The first script in the analysis is the main user script. Its node is used as
-        # the "caller" node for all others. This gives a connected graph rather than
-        # a collection of unrelated trees, one for each of self.inputs.
-        # The list of scripts is in self.inputs, each as a normalized pathname.
+
+        ### Hook cache.
+        logger.info('Caching module hooks...')
+
+        # List of all directories containing hook scripts. Default hooks are
+        # listed before and hence take precedence over custom hooks.
+        module_hook_dirs = [get_importhooks_dir()] + self.hookspath
+
+        # Lazily defer loading hook scripts until their first access.
```

Comment:
This does not "Lazily defer", but prepooulate the cache - which itself is in charge of lazy defer.